draft-ietf-core-stateless.xml

```diff
@@ -841,51 +841,94 @@
+        </t>
 
-          Where encryption is not needed, <xref
+        <t>
+          When encryption is used, the use of <xref
+          target="RFC3610">AES-CCM</xref> with a 64-bit tag is RECOMMENDED,
+          combined with a sequence number and a replay window.
+          This choice is informed by available hardware acceleration of on
+          many constrained systems.
+          If a different algorithm is available accelerated on the sender,
+          with similar strength, then it SHOULD be preferred.
+
+          Where privacy of the state is not required, and encryption is not needed, <xref
           target="RFC6234">HMAC-SHA-256</xref>, combined with a sequence number
-          and a replay window, may be used.
+          and a replay window, MAY be used.
         </t>
 
         <t>
-          When using an encryption mode that depends on a nonce, such as AES-CCM, repeated use of the same nonce under the same key
-          causes the cipher to fail catastrophically. If a nonce is ever used
-          for more than one encryption operation with the same key, then the
-          same key stream gets used to encrypt both plaintexts and the
+          When using an encryption mode that depends on a nonce, such as
+          AES-CCM, repeated use of the same nonce under the same key
+          causes the cipher to fail catastrophically.
+        </t>
+        <t>
+          If a nonce is ever used for more than one encryption operation with
+          the same key, then the same key stream gets used to encrypt both plaintexts and the
           confidentiality guarantees are voided. Devices with low-quality entropy
           sources -- as is typical with constrained devices, which incidentally
           happen to be a natural candidate for the stateless mechanism described
           in this document -- need to carefully pick a nonce generation
-          mechanism that provides the above uniqueness guarantee. Additionally,
-          since it can be difficult to use AES-CCM securely when using
-          statically configured keys, implementations should use <xref
-          target="RFC4107">automated key management</xref>.
+          mechanism that provides the above uniqueness guarantee.
+        </t>
+
+        <t>
+          A node which enters long-term sleeps may find it difficult to
+          maintain the state for a good pseudo-random number generated
+          appropriate for non-repeating nonce generation just for the
+          encryption of this state.
         </t>
 
+        <t>
+          If a node does not need to accept responses across sleep cycles,
+          then generating a new key for encryption of state during each wake
+          cycle is advised.
+        </t>
+        <t>
+          A node which does need to accept response between sleep cycles will
+          need to maintain the encryption key.  It may be stored into
+          available non-volatile storage once, provided that there is a
```

Comment:
once (e.g., at boot time)

(I don't think we want to fix the key at commissioning time, do we?)

Reply:
OSCORE APP B 1.1